doc/design/draft/named-segments.md

```diff
@@ -0,0 +1,219 @@
+
+In RouDi, given:
+* `userGroups` - A list of POSIX user groups a publishing process belongs to called 
+* `segmentName` - The (possibly empty) name of a shared memory segment specified in the publisher options call
+* `mappedSegments` - A list of shared memory segments
+
+1. If `segmentName` is not empty
+    1. Find the segment in `mappedSegments` matching the `segmentName`
+    2. If the containing process has write access to the segment via one of its `userGroups`, return the corresponding segment information
+    3. Otherwise return an error indicating the publisher does not have write access to the requested segment
+2. If it is empty:
+    1. Iterate over all process `userGroups`
+    2. Determine if the user group name matches the name of one of the segments
+    3. If it does:
+        1. If a match has already been previously made, return an error indicating that the publisher must only have write access to one segment
+        2. If not, verify the process has access to the segment and record the match
+    4. At the end of iteration, if a matching segment has been found, return the segment information
+    5. Otherwise return an error indicating that no matching segment has been found
+
+#### Runtime requesting segments to map
+
+In the client process, while initializing the runtime, given:
+* `userGroups` - A list of POSIX user groups a publishing process belongs to called 
+* `segmentContainer` - A list of shared memory segment information
+* `premappedSegments` - An enum indicating which group of segments should be mapped by default
+* `segmentFilter` - A (possibly empty) list of segment names to filter against
+
+In order to determine which segments to map:
+
+1. Map segments according to the `premappedSegments` option
+  1. If read access segments are requested, map every segment the user has read access to, but not write access
+  2. If write access segments are requested, map every segment the user has write access to
+  3. If read and write access segmeents are requested, map every segment the user has either read or write access to
+  4. If no segments are requested, do nothing
+2. If `segmentFilter` is not empty
+    1. Iterate over each name in the filter
+    2. If there is a segment that matches the name in the filter
+        1. If the segment has already been mapped, return an error indicating that a segment has been requested twice. The error should contain access permissions and the value of the `premappedSegments` option the user understands why it has already been mapped.
+        2. Otherwise, if the process has access rights to that segment, map it
+        2. If the process does not have access rights, return an error indicating that a segment was requested the runtime does not have access to.
+    3. If there is no segment that matches the name in the filter, return an error indicating that there is no segment matching the one requested to be mapped.
```

Comment:
Like already mentioned above. I have the feeling this becomes confusing due to the parameter influencing each other too much.

It might also break applications by just changing the segment config in a compatible way. Let's assume one want's to map all segments with write access with the `premappedSegments` options and one segment with read access from `segmentFilter`, let's call the segment `alice`. Now the config is changed so that the process has also write access to `alice` and suddenly the application does run anymore although it is not a breaking change in the configuration.

Reply:
Hm yeah. It could be pretty annoying even though you _could_ guard against it.

Another option is to simply ignore, but log a warning, about the case where the segment is already mapped. Most users wouldn't care anyway.